### ISSUE TITLE
LO allows pinning items and subclasses from other classes

### ISSUE THREAD

**Issue by robojumper** [DestinyItemManager/DIM#7593]
### DIM Version

Version 6.97.1.1001097 (beta), built on 29.12.2021, 10:30:37, also Version 6.97.1 (release), built on 27.12.2021, 03:21:55

### Browser Details

Firefox 95.0.2

### OS Details

Windows 10

### Describe the bug

I opened loadout optimizer on my Titan and clicked "pin item", then clicked Hunter while the pin item popup was still open. After switching to Hunter, I clicked on my One-Eyed Mask in the pin item popup. I expected LO to have closed or refreshed the picker when switching, or disagree with me in some way when clicking OEM, but instead LO pinned that OEM and optimized a Hunter loadout with OEM in the helmet slot.

You can do the same with the different stasis subclasses on beta, which is how I discovered this.

The exotic picker properly refreshes when switching classes.

### Logs

```shell
[loadout optimizer] Processing 20300 combinations from 84 items 
Object { helms: 1, gauntlets: 25, chests: 29, legs: 28, classItems: 1 }
```

**Comment by smerrill** [comment]
I was also looking at taking a pass at this, but I don't know how to approach this well (last time I used React was before hooks). I see that the item picker is brought up in two buttons with onClick handlers that invoke the `ItemPicker`, and since they're not component of their own, they don't re-render when props change. That doesn't matter anyway since it's the single global `ItemPicker` that lives outside the scope of the `LockArmorAndPerks` component.

Can I use the `EventBus` to tell the `ItemPicker` to close, or should I implement that?

I also tried re-calling `chooseLockItem()` or `chooseExcludeItem()` in a `useEffect()` hook, but to no avail.
